packages/metals-languageclient/src/fetchMetals.ts

```diff
@@ -19,17 +25,67 @@ interface PackedChildPromise {
   promise: ChildProcessPromise;
 }
 
-export async function fetchMetals(
-  {
-    serverVersion,
-    serverProperties,
-    javaConfig: { javaPath, javaOptions, extraEnv, coursierPath },
-  }: FetchMetalsOptions,
+export async function setupCoursier(
+  javaVersion: JavaVersion,
+  coursierFetchPath: string,
   output: OutputChannel
-): Promise<PackedChildPromise> {
-  const fetchProperties = serverProperties.filter(
-    (p) => !p.startsWith("-agentlib")
-  );
+): Promise<{ coursier: string; javaHome: string }> {
+  const handleOutput = (out: Buffer) => {
+    const msg = out.toString().trim();
+    output.appendLine("CS: " + msg);
+  };
+
+  const resolveCoursier = async () => {
+    const envPath = process.env["PATH"];
+    const defaultCoursier = path.resolve(coursierFetchPath, "cs");
+    const possibleCoursier: string | undefined = await validateCoursier(
+      envPath,
+      defaultCoursier
+    );
+
+    if (possibleCoursier) return possibleCoursier;
+
+    output.appendLine(`Fetching coursier`);
+
+    return fetchCoursier(coursierFetchPath, handleOutput)
+      .then(() => defaultCoursier)
+      .catch((err) => {
+        output.appendLine(
+          "Failed to fetch coursier. You may want to try installing coursier manually and adding it to PATH."
+        );
+        throw err;
+      });
+  };
+
+  const resolveJavaHome = async (coursier: string) => {
+    await run(
+      coursier,
+      ["java", "--jvm", javaVersion, "--setup"],
```

Comment:
Did you manage to check if https://get-coursier.io/docs/cli-java#system-jvm-detection is true? Looks like it doesn't work for me and even if JAVA_HOME is set to JDK 17 it will still try to download :thinking: 

Would be great to make sure we don't only rely on coursier, because would not manage to get Metals working when Java cannot be downloaded.

If coursier is at fault we may have to do a fallback or chekc manually beforehand

Reply:
I'll check manually. It seems to be the same for me, it downloads JDK 17  even if Java home is 17. I'm a bit confused by the docs, those don't really explain how it should behave when java version is set with respect to JAVA_HOME.

Reply:
This might be broken then :thinking: We might need to still use the find java home npm package based on the selected JDK version.